app/controllers/api/storage.php

```diff
@@ -1293,10 +1300,11 @@
             throw new Exception(Exception::STORAGE_BUCKET_NOT_FOUND);
         }
 
+        $isToken = !$resourceToken->isEmpty() && $resourceToken->getAttribute('bucketInternalId') === $bucket->getInternalId();
```

Comment:
The PR looks good, but I do have some doubts regarding the maintenance scalability of this security check. It feels very complex and very easy to get wrong, also, if we want to support more resources it will be very hard to confidently replicate. 

I wonder if we can abstract this in a higher level like DB authorization check or one of the init hooks as a fallback option.

Reply:
@eldadfux 
We could do this check beforehand around here - https://github.com/appwrite/appwrite/blob/fix-missing-filter/app/init/resources.php#L932C9-L953C6

That way it throws an error before we even hit the controller. WDYT?

Reply:
@eldadfux I think the checks needs to be present per endpoint, a helper method like logic could help. but I don't think using a hook would work because even if the token is valid, we still need to verify with an additional check like we do now to not throw the exceptions.

Reply:
cc @Meldiron @eldadfux as per internal conversations, planned for later.